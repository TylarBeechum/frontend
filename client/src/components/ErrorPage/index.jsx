import React from 'react';
import styled from 'styled-components';

const errorMsg = 'Oops, this is not the page you are looking for....';

// Subtract pixels to prevent unnecessary vertical scrollbar (accounts for fixed nav bar at the top and other elements)
const Container = styled.main`
  min-height: -webkit-calc(100vh - 200px);
  min-height: -moz-calc(100vh - 200px);
  min-height: calc(100vh - 200px);
  margin: 125px 7.5% 0;
`;

const ErrorCodeMsg = styled.h1`
  @keyframes errorPageHeadingExpand {
    from {
      transform: scale(0);
    }
    to {
      transform: scale(1);
    }
  }

  font-size: 4.5rem;
  animation: errorPageHeadingExpand 1s;

  @media (max-width: 500px) {
    font-size: 3.5rem;
  }
`;

const ErrorMsgHeading = styled.h3`
  @keyframes errorCodeFadeIn {
    from {
      opacity: 0;
    }
    to {
      opacity: 1;
    }
  }

  font-size: 2.25rem;
  margin: 25px 0 50px;
  animation: errorCodeFadeIn 1.5s 2s backwards;

  @media (max-width: 500px) {
    font-size: 2rem;
  }
`;

function LandingPage() {
  return (
    <Container>
      <ErrorCodeMsg>Code 404</ErrorCodeMsg>
      <ErrorMsgHeading>{errorMsg}</ErrorMsgHeading>
    </Container>
  );
}

export default LandingPage;
