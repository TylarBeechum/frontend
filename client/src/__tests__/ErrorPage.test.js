/* eslint-disable */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import {
  renderWithRouterAndProviders,
  rendererWithRouterAndProviders
} from '../__testHelpers__';

import ErrorPage from '../components/ErrorPage';

describe('<ErrorPage />', () => {
  describe('should render the component successfully without crashing', () => {
    it('tests that the component does not crash when it is rendered', () => {
      const helpers = renderWithRouterAndProviders(<ErrorPage />);
    });
  });

  describe('should match the previous snapshot of the component', () => {
    it('tests that the component has not changed and matched the previous snapshot', () => {
      const tree = rendererWithRouterAndProviders(<ErrorPage />);
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe('should render the title of the error page successfully to the screen', () => {
    it('tests that the title of the error page renders successfully', () => {
      const { getByText } = renderWithRouterAndProviders(<ErrorPage />);
      const title = getByText(/Code 404/i);
      expect(title).toBeDefined();
    });
  });

  describe('should render the error message to the screen successfully', () => {
    const { getByText } = renderWithRouterAndProviders(<ErrorPage />);
    const msg = getByText(
      /Oops, this is not the page you are looking for..../i
    );
    expect(msg).toBeDefined();
  });
});
