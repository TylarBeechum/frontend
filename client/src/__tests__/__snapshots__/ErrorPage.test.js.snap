// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorPage /> should match the previous snapshot of the component tests that the component has not changed and matched the previous snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 iDwiuh"
>
  <main
    className="sc-EHOje hExNRU"
  >
    <h1
      className="sc-bZQynM fMmKOo"
    >
      Code 404
    </h1>
    <h3
      className="sc-gzVnrw heKplq"
    >
      Oops, this is not the page you are looking for....
    </h3>
  </main>
</div>
`;
